### PR TITLE
Fix native build on Windows Arm64

### PR DESCRIPTION
**Pull request by Blackhex**
Tested on SciPy repository.

### CODE REVIEW

**Comment by xclaesse** [comment]
Someone did exactly the same thing and then disappeared, same comments applies: https://github.com/mesonbuild/meson/pull/11481

**Comment by Blackhex** [comment]
Thank you for you feedback, I'll test your proposed changes...

**Comment by xclaesse** [comment]
>         if mesonlib.is_windows() and self.get_windows_python_arch().endswith('64') and mesonlib.version_compare(self.version, '<3.12'):
>             self.compile_args += ['-DMS_WIN64=']

Note that I have no idea if `MS_WIN64` is needed for aarch64, maybe @eli-schwartz has an opinion on that.

File: mesonbuild/dependencies/python.py [hunk 1]
```diff
@@ -305,6 +305,8 @@ def find_libpy_windows(self, env: 'Environment') -> None:
             arch = '32'
         elif arch == 'x86_64':
             arch = '64'
+        elif arch == 'aarch64':
```
Comment (tristan957) [inline mesonbuild/dependencies/python.py:308]:
```suggestion
        elif arch in {'x86_64', 'aarch64'}:
```

Comment (tristan957) [reply mesonbuild/dependencies/python.py:308]: Combine the two branches into one.

**Comment by Blackhex** [comment]
The proposed changes seem to be working.

**Comment by xclaesse** [comment]
CI issue is unrelated. LGTM

**Pull request merged**